src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -125,8 +125,9 @@ private void Out(string s)
 
         private void OutMember(Expression instance, MemberInfo member, Type exprType)
         {
-            bool isId = false;
-            var name = GetPropertyName(member.Name, exprType);
+            var isId = false;
+            var name = _conventions.PropertyNameConverter(member) ?? GetPropertyName(member.Name, exprType);
```

Comment:
The convention is not null (or does not return null by default, which means that it changes the old behavior and introduces a breaking change.

Reply:
Good point, let's make it null / noop by default